### PR TITLE
Add conventions support for owned entity types.

### PR DESCRIPTION
**Pull request by AndriySvyryd**
InversePropertyAttribute is not supported.

### CODE REVIEW

File: build/common.props [hunk 1]
```diff
@@ -11,6 +11,7 @@
     <SignAssembly>True</SignAssembly>
     <TreatWarningsAsErrors>True</TreatWarningsAsErrors>
     <VersionSuffix Condition="'$(VersionSuffix)' != '' And '$(BuildNumber)' != ''">$(VersionSuffix)-$(BuildNumber)</VersionSuffix>
+    <DebugType Condition="'$(Configuration)' == 'Debug'">full</DebugType>
```
Comment (AndriySvyryd) [inline build/common.props]: @bricelam Is this ok?

Comment (bricelam) [inline build/common.props]: Hmm... Maybe add `AND '$(OS)' == 'Windows_NT'`

File: src/EFCore/Infrastructure/CoreModelValidator.cs [hunk 2]
```diff
@@ -203,14 +242,11 @@ protected virtual void EnsureFieldMapping([NotNull] IModel model)
...
                 {
-                    MemberInfo _;
-                    string errorMessage;
-
                     if (!propertyBase.TryGetMemberInfo(
                         forConstruction: true,
                         forSet: true,
-                        memberInfo: out _,
-                        errorMessage: out errorMessage))
+                        memberInfo: out MemberInfo _,
```
Comment (smitpatel) [inline src/EFCore/Infrastructure/CoreModelValidator.cs]: can use `var`

File: src/EFCore/Metadata/Conventions/Internal/ForeignKeyPropertyDiscoveryConvention.cs [hunk 3]
```diff
@@ -101,7 +108,7 @@ public virtual InternalRelationshipBuilder Apply(InternalRelationshipBuilder rel
                     foreignKey.PrincipalKey.Properties,
                     foreignKey.IsRequired,
                     foreignKey.DependentToPrincipal == null
-                        ? foreignKey.PrincipalEntityType.DisplayName() : foreignKey.DependentToPrincipal.Name);
+                        ? foreignKey.PrincipalEntityType.ShortName() : foreignKey.DependentToPrincipal.Name);
```
Comment (smitpatel) [inline src/EFCore/Metadata/Conventions/Internal/ForeignKeyPropertyDiscoveryConvention.cs]: `foreignKey.DependentToPrincipal?.Name ?? foreignKey.PrincipalEntityType.ShortName()`
Whichever you think is better.

File: src/EFCore/Metadata/Internal/InternalEntityTypeBuilder.cs [hunk 4]
```diff
@@ -1614,7 +1615,7 @@ private static IndexBuilderSnapshot DetachIndex(Index indexToDetach)
             [NotNull] string targetEntityTypeName,
             [NotNull] string navigationName,
             ConfigurationSource configurationSource)
-            => Owns(new TypeIdentity(targetEntityTypeName), PropertyIdentity.Create(navigationName), configurationSource);
+            => Owns(new TypeIdentity(targetEntityTypeName), PropertyIdentity.Create(navigationName), null, configurationSource);
```
Comment (smitpatel) [inline src/EFCore/Metadata/Internal/InternalEntityTypeBuilder.cs]: Can we have parameter name? in `/* */`

Comment (AndriySvyryd) [inline src/EFCore/Metadata/Internal/InternalEntityTypeBuilder.cs]: Even better, it can be not a comment

**Comment by AndriySvyryd** [comment]
@ajcvickers Would you like to review this as well?

**Comment by ajcvickers** [comment]
@AndriySvyryd I will if you want, but I'm happy with @smitpatel's sign off. He has good context on conventions. Are there behaviors that you think need reviewing?

**Comment by AndriySvyryd** [comment]
@ajcvickers The new exceptions. DIET names are formatted like this in the exception messages:
`Parent.Child -> Child`

File: src/EFCore.Specification.Tests/LoadTestBase.cs [hunk 5]
```diff
@@ -4605,6 +4605,9 @@ public abstract class LoadFixtureBase
 
             protected virtual void OnModelCreating(ModelBuilder modelBuilder)
             {
+                modelBuilder.Entity<SingleShadowFk>()
+                    .Property<int?>("ParentId");
+
```
Comment (ajcvickers) [inline src/EFCore.Specification.Tests/LoadTestBase.cs]: Why was this change needed?

File: test/EFCore.Tests/Infrastructure/CoreModelValidatorTest.cs [hunk 6]
```diff
@@ -253,6 +254,79 @@ public virtual void Detects_generic_leaf_type()
             VerifyError(CoreStrings.AbstractLeafEntityType(entityGeneric.DisplayName()), model);
         }
 
+        [Fact]
+        public virtual void Pases_on_valid_delegated_identity_entiy_types()
```
Comment (ajcvickers) [inline test/EFCore.Tests/Infrastructure/CoreModelValidatorTest.cs]: >entiy [](start = 62, length = 5)

Typo

File: src/EFCore/Properties/CoreStrings.resx [hunk 7]
```diff
@@ -735,4 +735,13 @@
...
+  <data name="NoDefiningNavigation" xml:space="preserve">
+    <value>There is no navigation '{navigation}' on '{definingEntityType}' that was used to define the entity type with delegated identity '{entityType}'</value>
+  </data>
+  <data name="MultipleOwnerships" xml:space="preserve">
+    <value>The entity type '{entityType}' is the target of multiple ownership relationships.</value>
+  </data>
+  <data name="NonDefiningOwnership" xml:space="preserve">
+    <value>The ownership navigation '{ownershipNavigation}' should be the same as the defining navigation '{definingNavigation}' for entity type '{entityType}'</value>
+  </data>
 </root>
```
Comment (ajcvickers) [inline src/EFCore/Properties/CoreStrings.resx]: Related to comment on previous PR, we need to come up with some language to talk about this that will be understandable. Probably a design meeting thing. /cc @divega

File: src/EFCore.Specification.Tests/LoadTestBase.cs (hunk 5 above)
Comment (AndriySvyryd) [inline src/EFCore.Specification.Tests/LoadTestBase.cs]: The property created by default is not nullable, I think a nullable property is a more interesting test case.

Comment (AndriySvyryd) [inline src/EFCore.Specification.Tests/LoadTestBase.cs]: The property created by convention is now not nullable if the FK is required, but these tests still need a nullable property.
